k8s/k8s.go

```diff
@@ -316,6 +451,27 @@ func overrideIPAMResult(ipAddrsNoIpam string, logger *log.Entry) (*types.Result,
 	return &result, nil
 }
 
+// parseIPAddrs is a utility function that parses string of IPs in json format that are
+// passed in as a string and returns a slice of string with IPs.
+// It also makes sure the slice isn't empty.
+func parseIPAddrs(ipAddrsStr string, logger *log.Entry) ([]string, error) {
+	var ips []string
+
+	err := json.Unmarshal([]byte(ipAddrsStr), &ips)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse '%s' as JSON: %s", ipAddrsStr, err)
+	}
+
+	// annotation value can't be empty.
```

Comment:
Minor thing, but I'd recommend moving this `len()` check out of this function.  There's nothing about this helper that says it needs to be parsing an annotation, and it's really the calling code that knows if `len() == 0` is an error or not.